dbms/src/WindowFunctions/IWindowFunction.h

```diff
@@ -45,8 +45,25 @@ class IWindowFunction
 
 protected:
     DataTypes argument_types;
+
+private:
+    // Return true when the frame is valid
+    //
+    // Check if the frame is invalid, or we will insert null value into result column
+    // Frame is invalid when:
+    //   1. partition_end <= frame_start
+    //   2. frame_end < partition_start
+    virtual bool checkFrameValidAndHandle(WindowTransformAction &, size_t) { return true; }
```

Comment:
Will this function be called for all impl of  IWindowFunction, if not it seems not neccessary to add a top level virtual function here

Reply:
> Will this function be called for all impl of IWindowFunction, if not it seems not neccessary to add a top level virtual function here

okk, I will make it as a static function in .cpp file.